src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -45,7 +46,8 @@ void RandomFourierGaussPreproc::init_basis(int32_t dim_input_space)
 void RandomFourierGaussPreproc::fit(std::shared_ptr<Features> f)
 {
 	auto num_features = f->as<DenseFeatures<float64_t>>()->get_num_features();
-	require(num_features > 0, "Input space dimension must be greater than zero");
+	require(num_features > 0, "Given input space dimension {} must be positive", num_features);
```

Comment:
Minor nit: "Given input space dimension" -> "Dimension of provided features"
And also you should assert for positive number of features.
@gf712 I think we should check these things in the base class just like in Machine ... eventually

Reply:
Isn't the require method asserting that anyway? (with the added benefit of throwing error in case of failure)

Reply:
sorry that was a typo, I meant positive number of *examples*

Reply:
(and also that `f` is not nullptr)
But I think leave it for now ... we would do this in the base class ... later

Reply:
I think f not being a null is filtered out in num_features > 0  condition

Reply:
nope

Reply:
but no need to change. We will put that in the base class in some other PR